### ISSUE TITLE
Upload fails with time zone problem

### ISSUE THREAD

**Issue by cuh7b5** [cpfair/tapiriik#198]
I try to sync .tcx files of stationary workouts (i.e. without GPS data) from Dropbox to Endomondo, Runkeeper and Garmin Connect. The sync fails with these messages:

Endomondo: Endomondo does not support non-GPS activities with other sensor data.
Runkeeper: The time zone of this activity could not be determined.
Garmin Connect: The time zone of this activity could not be determined. 

I understand that Endomondo cannot accept such files but what is wrong with the GC and RK sync?

The .tcx files are standard ones. I even generated one with Garmin Connect to be as standard as it can be, and it gives the same error. I believe that the dates are correct with the mark "Z" showing that they are UTC times (e.g: <Time>2016-01-05T17:03:14.000Z</Time>  ). I can manually upload these files to GC with no issue. I attach one of them.

[activity_1012498591.tcx.zip](https://github.com/cpfair/tapiriik/files/89076/activity_1012498591.tcx.zip)

Anyone can help me with this? Thx

**Comment by svdberg** [comment]
Probably Tapiriik uses other information from the services (for example RunKeeper should provide an utc_offset field in their API) to recalculate things to UTC/Local time. My guess is that in the gpx parsing code tries to check in which timezone the GPS coordinates are, and because they are not there, fails to set the TZ info. I'll have a look at the code, because now you made me curious.

**Comment by cuh7b5** [comment]
Yes, I think this is what happening. But I don't really understand why the code wants to determine the timezone by the GPS info when this information is there in the tcx file (i.e. the Z mark at the end of the date.

**Comment by Pi-age** [comment]
I seem to sync to suffer from the same timezone problem when i try to sync a tacx HRM file that was converted by tcx creator to tcx from DB to GC. Any news on how to resolve the issue?

**Comment by cuh7b5** [comment]
No news. :( Stationary workouts are not uploaded anywhere due to a so called timezone problem although the workout contains standard and correct timezone information.

**Comment by theHasp** [comment]
any news? I wanted to sync stationary data from dropbox to velohero and garmin .... and got this error
